### PR TITLE
[minor_change] Add resource and datasource for mso_remote_location

### LINKED ISSUE #192: Add `mso_remote_location` resource and data source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add a new resource and data source to manage MSO remote locations.

### PR DESCRIPTION
**Pull request by akinross**
fixes #192

### CODE REVIEW

**Comment by akinross** [comment]
Still requires mso-go-client dependency update to be added before merge, waiting on review and approval.

File: mso/resource_mso_remote_location.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+			return nil
+		},
+	}
+}
+
+func setAuthenticationInState(d *schema.ResourceData) {
+
+	d.Set("password", "")
+	d.Set("ssh_key", "")
+	d.Set("passphrase", "")
```
Comment (sajagana) [inline mso/resource_mso_remote_location.go:110]: This may cause an issue when store_in_statefile is true.

**Review by sajagana — CHANGES_REQUESTED** [review]

File: mso/resource_mso_remote_location.go (hunk 1 above)
Comment (akinross) [reply mso/resource_mso_remote_location.go]: you are correct should be in conditional, changed it

File: website/docs/r/remote_location.html.markdown [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+## Attribute Reference ##
+
+The only attribute exported with this resource is `id`. Which is set to the id of user associated.
+
+## Importing ##
+
+An existing MSO User can be [imported][docs-import] into this resource via its Id, via the following command: [docs-import]: <https://www.terraform.io/docs/import/index.html>
+
+```bash
+terraform import mso_user.mso_remote_location {remote-location-id}
```
Comment (anvitha-jain) [inline website/docs/r/remote_location.html.markdown:79]: Are you sure this is right?

Shouldn't it be like terraform import mso_remote_location.example {remote-location-id}

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by anvitha-jain — DISMISSED** [review]

File: mso/resource_mso_remote_location.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  22,
+			},
+			"username": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"password": &schema.Schema{
```
Comment (shrsr) [inline mso/resource_mso_remote_location.go:68]: Was password intentionally set to computed?

File: mso/resource_mso_remote_location.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+			},
+		}),
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
+			_, password_ok := diff.GetOk("password")
+			_, ssh_key_ok := diff.GetOk("ssh_key")
+			if password_ok && ssh_key_ok {
+				return errors.New(`"password" and "ssh_key" cannot be provided in the same resource.`)
+			}
+			if !password_ok && !ssh_key_ok {
+				return errors.New(`"password" or "ssh_key" must be provided in the same resource.`)
```
Comment (shrsr) [inline mso/resource_mso_remote_location.go:93]: Suggestion -
"password" or "ssh_key" is required to manage the MSO remote location.

File: website/docs/d/remote_location.html.markdown [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+# mso_user #
+
+Data source for MSO Remote Location
+
+## Example Usage ##
+
+```hcl
+
+data "mso_remote_location" "password" {
+  name = "remote_location_password"
```
Comment (shrsr) [inline website/docs/d/remote_location.html.markdown:18]: name = "remote_location_password" -> "remote_location"

Comment (shrsr) [reply website/docs/d/remote_location.html.markdown:18]: This was a little confusing to read at first

**Review by shrsr — CHANGES_REQUESTED** [review]

**Comment by lhercot** [comment]
Need a rebase

File: mso/resource_mso_remote_location.go (hunk 3 above)
Comment (akinross) [reply mso/resource_mso_remote_location.go:68]: Yes is there a reason it shouldn't?

File: mso/resource_mso_remote_location.go (hunk 4 above)
Comment (akinross) [reply mso/resource_mso_remote_location.go:93]: ok

File: website/docs/d/remote_location.html.markdown (hunk 5 above)
Comment (akinross) [reply website/docs/d/remote_location.html.markdown:18]: what do you mean? I kept different names for the resource on purpose, what name would you propose?

Comment (shrsr) [reply website/docs/d/remote_location.html.markdown:18]:
```
data "mso_remote_location" "password" {
    name = "remote_location_name"
}
```

File: mso/resource_mso_remote_location.go (hunk 3 above)
Comment (shrsr) [reply mso/resource_mso_remote_location.go:68]: It would be good to add **Sensitive: True** here

**Review by shrsr — CHANGES_REQUESTED** [review]

File: mso/resource_mso_remote_location.go (hunk 3 above)
Comment (akinross) [reply mso/resource_mso_remote_location.go:68]: ok added for password, sshkey and passphrase

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**